client/dashboard/me/security-account-recovery/recovery-email.tsx

```diff
@@ -0,0 +1,220 @@
+import type { Field } from '@wordpress/dataviews';
+
+type SecurityEmailFormData = {
+	email: string;
+};
+
+export default function RecoveryEmail() {
+	const [ formData, setFormData ] = useState< SecurityEmailFormData >( {
+		email: '',
+	} );
+	const [ isRemoveDialogOpen, setIsRemoveDialogOpen ] = useState( false );
+	const [ showResendButton, setShowResendButton ] = useState( true );
+
+	const { data: accountRecoveryData, isLoading: isAccountRecoveryDataLoading } = useQuery(
+		accountRecoveryQuery()
+	);
+
+	const { mutate: validateEmailMutation, isPending: isValidateEmailPending } = useMutation(
+		updateAccountRecoveryEmailMutation()
+	);
+
+	const { mutate: removeEmailMutation, isPending: isRemoveEmailPending } = useMutation(
+		removeAccountRecoveryEmailMutation()
+	);
+
+	const { mutate: resendValidationMutation, isPending: isResendValidationPending } = useMutation(
+		resendAccountRecoveryEmailValidationMutation()
+	);
+
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const accountRecoveryEmail = accountRecoveryData?.email;
+
+	useEffect( () => {
+		setFormData( { email: accountRecoveryEmail || '' } );
+	}, [ accountRecoveryEmail ] );
+
+	const handleSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+		validateEmailMutation( formData.email, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Your recovery email was saved successfully.' ), {
+					type: 'snackbar',
+				} );
+			},
+			onError: ( error: Error ) => {
+				createErrorNotice( error.message || __( 'Failed to save recovery email.' ), {
+					type: 'snackbar',
+				} );
+			},
+		} );
+	};
+
+	const handleRemove = () => {
+		setIsRemoveDialogOpen( false );
+		removeEmailMutation( undefined, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Your recovery email was removed successfully.' ), {
+					type: 'snackbar',
+				} );
+				setFormData( { email: '' } );
+			},
+			onError: ( error: Error ) => {
+				createErrorNotice( error.message || __( 'Failed to remove recovery email.' ), {
+					type: 'snackbar',
+				} );
+			},
+		} );
+	};
+
+	const handleResendValidation = () => {
+		setShowResendButton( false );
+		resendValidationMutation( undefined, {
+			onSuccess: () => {
+				createSuccessNotice( __( 'Your recovery email validation was resent successfully.' ), {
+					type: 'snackbar',
+				} );
+			},
+			onError: ( error: Error ) => {
+				createErrorNotice( error.message || __( 'Failed to resend recovery email validation.' ), {
+					type: 'snackbar',
+				} );
+			},
+		} );
+	};
+
+	const { data: serverData } = useQuery( profileQuery() );
+
+	const shouldShowValidationNotice = accountRecoveryEmail && ! accountRecoveryData?.email_validated;
+
+	const fields: Field< SecurityEmailFormData >[] = useMemo(
+		() => [
+			{
+				id: 'email',
+				label: __( 'Email address' ),
+				description:
+					/* translators: %s: email address */
+					__( 'Your primary email address is %s', serverData?.user_email ),
```

Comment:
I noticed that the description is not getting rendered. I'll debug this separately.